module/Kanbanize/src/Kanbanize/Controller/BoardsController.php

```diff
@@ -0,0 +1,203 @@
+			$error->addSecondaryErrors("streamName", ["Stream name cannot be empty"]);
+		}
+		if($error->hasErrors()){
+			$error->setCode(400);
+			$error->setDescription('Some parameters are not valid');
+			$this->response->setStatusCode(400);
+			return $error;
+		}
+		$filters = new FilterChain();
+		$filters->attach(new StringTrim())
+			->attach(new StripNewlines())
+			->attach(new StripTags());
+		$streamName = $filters->filter($data['streamName']);
+		$projectId = $data['projectId'];
+
+		$statusValidator = new InArray([
+			'haystack' => [
+					Task::STATUS_IDEA,
+					Task::STATUS_OPEN,
+					Task::STATUS_ONGOING,
+					Task::STATUS_COMPLETED,
+					Task::STATUS_ACCEPTED,
+					Task::STATUS_CLOSED
+			]
+		]);
+		foreach($data['mapping'] as $column=>$status){
+			if(!$statusValidator->isValid($status)){
+				$error->addSecondaryErrors($column, ["Invalid status: {$status}"]);
+			}
+		}
+		if($error->hasErrors()){
+			$error->setCode(400);
+			$error->setDescription('Some parameters are not valid');
+			$this->response->setStatusCode(400);
+			return $error;
+		}
+
+		$organization = $this->getOrganizationService()->getOrganization($this->organization->getId());
+		$kanbanizeSettings = $organization->getSetting(Organization::KANBANIZE_KEY_SETTING);
+		$kanbanizeSettings['boards'][$id]['columnMapping'] = $data['mapping'];
+		$stream = $this->kanbanizeService->findStreamByBoardId($id, $organization);
+		$this->transaction()->begin();
+		try{
+			if(is_null($stream)){
+				$stream = $this->createStream($streamName, $projectId, $id, $organization);
+			}else if($stream->getSubject() != $streamName){
+				$stream->setSubject($streamName, $this->identity());
+			}
+			$organization->setSettings(Organization::KANBANIZE_KEY_SETTING, $kanbanizeSettings, $this->identity());
+			$this->transaction()->commit();
+			$this->response->setStatusCode(201);
+			return new JsonModel([
+				'streamName' => $stream->getSubject(),
+				'boardId' => $id,
+				'boardSettings' => $organization->getSetting(Organization::KANBANIZE_KEY_SETTING)['boards'][$id]
+			]);
+		}catch (InvalidArgumentException $ex){
+			$this->transaction()->rollback();
+			$this->response->setStatusCode(422);
+			$error->setCode(ErrorJsonModel::$ERROR_INPUT_VALIDATION);
+			$error->setDescription($e->getMessage());
+			return $error;
+		}
+	}
+	
+	public function get($id){
+		if(is_null($this->identity())) {
+			$this->response->setStatusCode(401);
+			return $this->response;
+		}
+		if(!$this->isAllowed($this->identity(), $this->organization, 'Kanbanize.BoardSettings.get')) {
+			$this->response->setStatusCode(403);
+			return $this->response;
+		}
+		$organization = $this->getOrganizationService()->getOrganization($this->organization->getId());
+		$kanbanizeSettings = $organization->getSetting(Organization::KANBANIZE_KEY_SETTING);
+		$error = new ErrorJsonModel();
+		try{
+			$this->initApi($kanbanizeSettings['apiKey'], $kanbanizeSettings['accountSubdomain']);
```

Comment:
Fai ritornare dalla `initApi` l'oggetto inizializzato e non usare dentro questo metodo la variabile membro `$this->client` ma una variabile locale `$client = $this->initApi(...)`